back/app/services/trending_idea_service.rb

```diff
@@ -2,12 +2,12 @@
 
 class TrendingIdeaService
   def filter_trending(ideas = Idea.all)
-    with_trending_score(ideas)
+    with_trending_score(ideas.not_from_survey)
       .where('sub.is_trending' => true)
   end
 
   def sort_trending(ideas = Idea.all)
-    with_trending_score(ideas)
+    with_trending_score(ideas.not_from_survey)
```

Comment:
The bugfix to prevent including native survey response ideas that would included in the before hook count of ideas used to decide whether to send the mail, then later removed from the payload.

Reply:
I wanted to use a scope to make use of the querying mechanism, rather than using a model method and then iterating over a collection, as I thought this would be more efficient.

Is my thinking wrong?

Reply:
It makes sense, but
1. We already do it in other places https://github.com/CitizenLabDotCo/citizenlab/blob/efd999d61d004d12c250807233222f755fad0a16/back/engines/free/email_campaigns/app/models/email_campaigns/campaigns/admin_digest.rb#L171
2. If you still want to use a scope, maybe you could make more explicit that these two pieces of logic are related. E.g., naming the scope `include_data_in_email` instead of `not_from_survey`. 

Or maybe @sebastienhoorens has some better ideas on it.

Reply:
OK, I'll switch to using select with the existing model method `include_data_in_email?` 👍

Reply:
I would suggest using the `IdeasFinder`. I'm not a fan of the `IdeasFinder` but this is what we've used to filter out survey responses from the input manager. So using it would make it more consistent. In general it's important not to express conditionals like "not native survey" as we want to keep inputs as generic as possible.

Reply:
Great 'idea', @sebastienhoorens , and done!